File: briar-headless/src/main/java/org/briarproject/briar/headless/contact/OutputContact.java
package org.briarproject.briar.headless.contact;

import org.briarproject.bramble.api.contact.Contact;
import org.briarproject.bramble.api.nullsafety.NotNullByDefault;
import org.briarproject.bramble.identity.OutputAuthor;

import javax.annotation.concurrent.Immutable;

@Immutable
@NotNullByDefault
@SuppressWarnings("WeakerAccess")
public class OutputContact {

	public final int id;
	public final OutputAuthor author;
	public final boolean verified;

	public OutputContact(Contact c) {
		this.id = c.getId().getInt();
		this.author = new OutputAuthor(c.getAuthor());
		this.verified = c.isVerified();
	}

}

File: briar-headless/src/main/java/org/briarproject/briar/headless/contact/OutputContact.kt
package org.briarproject.briar.headless.contact

import org.briarproject.bramble.api.contact.Contact
import org.briarproject.bramble.api.contact.event.ContactAddedEvent
import org.briarproject.bramble.api.plugin.event.ContactConnectedEvent
import org.briarproject.bramble.api.plugin.event.ContactDisconnectedEvent
import org.briarproject.bramble.identity.output
import org.briarproject.briar.headless.json.JsonDict

internal fun Contact.output(latestMsgTime: Long, connected: Boolean, unreadCount: Int) = JsonDict(
    "contactId" to id.int,
    "author" to author.output(),
    "verified" to isVerified,
    "lastChatActivity" to latestMsgTime,
    "connected" to connected,
    "unreadCount" to unreadCount
).apply {
    alias?.let { put("alias", it) }
    handshakePublicKey?.let { put("handshakePublicKey", it.encoded) }
}

internal fun ContactAddedEvent.output() = JsonDict(
    "contactId" to contactId.int,
    "verified" to isVerified
)

internal fun ContactConnectedEvent.output() = JsonDict(
    "contactId" to contactId.int
)

internal fun ContactDisconnectedEvent.output() = JsonDict(
    "contactId" to contactId.int
)
